Add troubleshooting section for https (#754)
runatlantis/atlantis: runatlantis.io/docs/troubleshooting-https.md

# HTTPS, SSL, TLS

When using a self-signed certificate for Atlantis (with flags `--ssl-cert-file` and `--ssl-key-file`),
there are a few considerations.

Atlantis uses the web server from the standard Go library, 
the method name is [ListenAndServeTLS](https://golang.org/pkg/net/http/#ListenAndServeTLS).

> For info, ListenAndServeTLS acts identically to [ListenAndServe](https://golang.org/pkg/net/http/#ListenAndServe),

Reviewer: ```suggestion
`ListenAndServeTLS` acts identically to [ListenAndServe](https://golang.org/pkg/net/http/#ListenAndServe),
```

except that it expects HTTPS connections. 
Additionally, files containing a certificate and matching private key for the server must be provided. 
If the certificate is signed by a certificate authority, 

> the certFile should be the concatenation of the server's certificate, any intermediates, and the CA's certificate. 

Reviewer: ```suggestion
the file passed to `--ssl-cert-file` should be the concatenation of the server's certificate, any intermediates, and the CA's certificate. 
```

If you have this error when specifying a TLS cert with a key: 
```
[EROR] server.go:413 server: Tls: private key does not match public key
```

> Check that the locally signed certificate is prepended to the self signed certificate.

Reviewer: ```suggestion
Check that the locally signed certificate authority is prepended to the self signed certificate.
```
You mean authority right?
